### PR TITLE
Rocketeer 3.0.0

### PR DESCRIPTION
**Pull request by Anahkiasen**
## Features
- [ ] Refactor the way phpseclib is used by the Connection classes to allow the reading of prompts
- [ ] Add support for agent forwarding
- [ ] Refactor connections to allow parallelization of tasks themselves (and not just deployments)

## Changes
- [x] Remove illuminate/remote completely and use phpseclib directly (since illuminate/remote is deprecated)

## Frameworks
- [x] Move Laravel related classes into a separate package
- [ ] Add support or Laravel 5

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,37 @@
...
+    - PREFER_LOWEST="--prefer-lowest"
+    - PREFER_LOWEST=""
 
-matrix:
-  allow_failures:
-    - php: hhvm-nightly
-  fast_finish: true
+# Dependencies
+install:
+  - composer self-update
```
Comment (GrahamCampbell) [inline .travis.yml]: This should be done in before_install.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,51 @@
 # CHANGELOG
 
-2.0.4 - 2014-12-08
-------------------
+Rocketeer follows the [Semantic Versioning 2.0](http://semver.org/spec/v2.0.0.html) spec.
+
+## [2.1.0] - Unreleased
```
Comment (GrahamCampbell) [inline CHANGELOG.md]: 3.0.0 surely?

File: CHANGELOG.md [hunk 3]
```diff
@@ -357,3 +373,33 @@
 - Added `deploy:cleanup` command
 - Added config file
 - Added `deploy:setup` and `deploy:deploy` commands
+
+[2.1.0]: https://github.com/rocketeers/rocketeer/compare/2.0.4...develop
```
Comment (GrahamCampbell) [inline CHANGELOG.md]: 3.0.0

File: composer.json [hunk 4]
```diff
@@ -60,7 +62,7 @@
   },
   "extra": {
     "branch-alias": {
-      "dev-develop": "2.0-dev"
+      "dev-develop": "2.1-dev"
```
Comment (GrahamCampbell) [inline composer.json]: 3.0?

**Comment by Anahkiasen** [comment]
Closing cause this be messin' with mah Travis builds

**Pull request closed without merging**